src/context/WalletProvider/KeepKey/hooks/useKeepKeyCancel.ts

```diff
@@ -0,0 +1,36 @@
+import { useToast } from '@chakra-ui/react'
+import { useCallback } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { useHistory } from 'react-router-dom'
+import { WalletActions } from 'context/WalletProvider/actions'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+export const useKeepKeyCancel = () => {
+  const history = useHistory()
+  const toast = useToast()
+  const translate = useTranslate()
+  const {
+    state: { wallet },
+    dispatch,
+  } = useWallet()
+
+  const cancelWalletRequests = useCallback(async () => {
```

Comment:
Total nitpick, but as we can only ever cancel the current request, this should be named in the singular: `cancelWalletRequest`.